Add code samples for distributed lock (#2610)
dapr/docs: daprdocs/content/en/developing-applications/building-blocks/distributed-lock/howto-use-distributed-lock.md

description: "Learn how to use distributed locks to provide exclusive access to a resource"
---

Now that you've learned what the Dapr distributed lock API building block provides, learn how it can work in your service. The example below describes an application that aquires a lock. This example uses the Redis lock component to demonstrate how to lock resources. For a list of supported lock stores, see [this list](/reference/components-reference/supported-locks/).

> <img src="/images/building-block-lock-example.png" width=1000 alt="The diagram below shows two instances of the same application acquiring a lock, where one instance is successful and the other is denied">

Reviewer: ```suggestion
The diagram below shows two instances of the same application acquiring a lock, where one instance is successful and the other is denied.

<img src="/images/building-block-lock-example.png" width=1000 alt="The diagram below shows two instances of the same application acquiring a lock, where one instance is successful and the other is denied">
```

description: "Learn how to use distributed locks to provide exclusive access to a resource"
---

Now that you've learned what the Dapr distributed lock API building block provides, learn how it can work in your service. The example below describes an application that aquires a lock. This example uses the Redis lock component to demonstrate how to lock resources. For a list of supported lock stores, see [this list](/reference/components-reference/supported-locks/).

> <img src="/images/building-block-lock-example.png" width=1000 alt="Diagram showing aquiring a lock from multiple instances of same application">

Reviewer: ```suggestion
The diagram below shows two instances of the same application acquiring a lock, where one instance is successful and the other is denied.

<img src="/images/building-block-lock-example.png" width=1000 alt="Diagram showing acquiring a lock from multiple instances of same application">
```

Author: Changed.

> <img src="/images/building-block-lock-unlock-example.png" width=1000 alt="Diagram showing releasing a lock from multiple instances of same application">

Reviewer: ```suggestion
The diagram below shows two instances of the same application, where one instance releases the lock and the other instance is then able to acquire the lock.

<img src="/images/building-block-lock-unlock-example.png" width=1000 alt="Diagram showing releasing a lock from multiple instances of same application">
```

    value: <PASSWORD>
```

> ### Acquire Lock

Reviewer: ```suggestion
### Acquire lock
```

Author: Changed

{{< tabs HTTP Dotnet Go >}}

{{% codetab %}}

```bash
curl -X POST http://localhost:3500/v1.0-alpha1/lock/lockstore
   -H 'Content-Type: application/json'
   -d '{"resourceId":"my_file_name", "lockOwner":"random_id_abc123", "expiryInSeconds": 60}'
```

{{% /codetab %}}

{{% codetab %}}

```csharp
using System;
using Dapr.Client;

namespace LockService
{
    class Program
    {
        static async Task Main(string[] args)
        {
            string DAPR_LOCK_NAME = "lockstore";
            var client = new DaprClientBuilder().Build();

            using (var fileLock = await client.Lock(DAPR_LOCK_NAME, "my_file_name", "random_id_abc123", 60))
            {
                if (fileLock.Success)
                {
                    Console.WriteLine("Success");
                }
                else
                {
                    Console.WriteLine($"Failed to lock {fileName}.");
                }
            }
        }
    }
}
```

{{% /codetab %}}

{{% codetab %}}

```go
package main

import (
    "fmt"

    dapr "github.com/dapr/go-sdk/client"
)

func main() {
    client, err := dapr.NewClient()
    if err != nil {
        panic(err)
    }
    defer client.Close()
    

>     resp, err := client.TryLockAlpha1(ctx, testLockStore, &dapr.LockRequest{

Reviewer: ```suggestion
    resp, err := client.TryLockAlpha1(ctx, "lockstore", &dapr.LockRequest{
```
Also, why is this method called Alpha1? Will have to change this in the future. Could this be "TryLock"

Author: No, it can't be `TryLock`. The Go-SDK uses Go semantics that prefer explicit naming conventions to express Alpha APIs. We've been doing the same for the Query and others.

> 			OwnerID:         "random_id_abc123",

Reviewer: Why is this OwnerID as opposed to LockOwner? Can this be changed?

Author: Yes, this is changed now.

			ResourceID:      "my_file_name",
			ExpiryInSeconds: 60,
		})

    fmt.Println(resp.Success)
}
```

{{% /codetab %}}

{{< /tabs >}}

> ### Unlock Existing Lock

Reviewer: ```suggestion
### Unlock existing lock
```

Author: Changed
